### ISSUE TITLE
[Enhancement]: Battle Shout Buff Notification Fix / Multiple Warriors in Party

### ISSUE THREAD

**Issue by Soulsmacka** [herotc/hero-rotation#958]
### What feature would you like to have added?

This has been an ongoing issue since I started using this addon. 

If I'm in a party or raid with another Warrior and they cast Battle Shout, the addon suggests I need battle shout even if I still have the buff. 

I think this also happens with other buffs like stamina buffs on other classes,

I'm assuming that the simplest fix is to code the addon to just see if the buff already exists and not recommend it.  

If this is something that can't be coded in for whatever reason, Is it possible that a missing buff, that players can provide themselves, can have its own unique spot rather than appear in the CD window similar to how you have a unique spot for Odyn's Fury, but please, do not start using that same spot to notify about not having battle shout and it starts blocking when Odyn's Fury is available.  If this is something that requires its own spot, then please create it and take it out of the CD area. I'm sure all the other classes that provide buffs will appreciate this QoL update. 

I know when another warrior is also using your addon because we both start weaving battle shout into our rotation during a fight. I literally need to replace slam sometimes with battle shout just so I have the key bind accessible for my rotation. And we just go bank n forth until one of us gives up.

**Comment by Cilraaz** [comment]
You did not mention what spec is having this issue.

The addon is coded as you suggest. It currently checks to see if Battle Shout is castable and either the buff is not active on the player (casted by anyone) or if anyone in the party/raid (within 100 yards) is missing the buff.

**Comment by Soulsmacka** [comment]
It happens in all warrior specs. Is there a way to get it out of the cool-down window? Can you create a missing buff window for the addon?

**Comment by Soulsmacka** [comment]
> You did not mention what spec is having this issue.
> 
> The addon is coded as you suggest. It currently checks to see if Battle Shout is castable and either the buff is not active on the player (casted by anyone) or if anyone in the party/raid (within 100 yards) is missing the buff.

It happens in all warrior specs. Is there a way to get it out of the cool-down window? Can you create a missing buff window for the addon?
And please, do not use the Odyn's fury window. it's perfect and don't need the notification for the missing battle shout to appear there.

**Comment by Cilraaz** [comment]
There is an option to move Battle Shout to the OffGCD area (Options -> AddOns -> HeroRotation -> Warrior).

I'll do some testing to see if I can figure out why the addon would be suggesting Battle Shout when the buff is up. Keep in mind that it does check everyone in your party/raid. So if someone was out of range and then enters range or dies and is rezzed, it will be suggested again.

**Comment by Cilraaz** [comment]
I did some testing with another Warrior. We did not have any issues with the addon acting as expected. Either player's Battle Shout counted as a valid Battle Shout, and while the other player's Battle Shout was active, it was not suggested for me to use Battle Shout.

I will add a toggle setting to show or suppress Battle Shout suggestions during combat. At minimum, that will make it so users who do not want to see the suggestion mid-combat can suppress them.
